syntax = "proto3";

package common;

// Common message types used across all services

// Health check messages
message HealthCheckRequest {
  string service = 1;
}

message HealthCheckResponse {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
    SERVICE_UNKNOWN = 3;
  }
  ServingStatus status = 1;
}

// Common error response
message ErrorResponse {
  int32 code = 1;
  string message = 2;
  string details = 3;
}

// Audio data format
message AudioData {
  bytes data = 1;
  int32 sample_rate = 2;
  int32 channels = 3;
  string format = 4; // "wav", "pcm", etc.
}

// Text data with metadata
message TextData {
  string text = 1;
  string language = 2;
  float confidence = 3;
}

// Session information
message SessionInfo {
  string session_id = 1;
  string conversation_id = 2;
  int64 timestamp = 3;
}

// Service status
message ServiceStatus {
  string service_name = 1;
  bool is_healthy = 2;
  string version = 3;
  int64 uptime_seconds = 4;
  map<string, string> metadata = 5;
}